src/outSideFunction/functionCart/Payment.js

```diff
@@ -0,0 +1,29 @@
+import axios from "../../axiosConfig";
+import { POST } from "../../constans/AxiosPost";
+
+const usePayment = () => {
+  const payment = async (valContext) => {
+    if (valContext.selectedIteamToPay.length !== 0) {
+      try {
+        const res = await axios.post(POST.PAY, {
+          items: valContext.selectedIteamToPay,
+          userId: valContext.userData._id,
+        });
+        const data = await res.data;
+        if (data === "Payment successful") {
+          alert(data);
+          valContext.selectedIteamToPay.forEach((valProductSelected) => {
+            valContext.paymentCart(valProductSelected);
+          });
+        }
+      } catch (error) {
+        console.log(error);
+      }
+    } else {
+      alert(`select a product`);
+    }
+  };
+
+  return { payment };
+};
```

Comment:
Same comment here about the function name, payment sounds a lot like variable name for me